packages/node/src/indexer/unfinalizedBlocks.service.ts

```diff
@@ -22,13 +30,119 @@ export function blockToHeader(block: BlockContent | Block): Header {
 }
 
 @Injectable()
-export class UnfinalizedBlocksService extends BaseUnfinalizedBlocksService<BlockContent> {
+export class UnfinalizedBlocksService extends BaseUnfinalizedBlocksService<EthereumBlock> {
+  private supportsFinalization?: boolean;
+  private startupCheck = true;
+
   constructor(
     private readonly apiService: ApiService,
     nodeConfig: NodeConfig,
     storeCache: StoreCacheService,
   ) {
-    super(nodeConfig, storeCache);
+    super(new EthereumNodeConfig(nodeConfig), storeCache);
+  }
+
+  /**
+   * @param reindex - the function to reindex back before a fork
+   * @param supportsFinalization - If the chain supports the 'finalized' block tag this should be true.
+   * */
+  async init(
+    reindex: (targetHeight: number) => Promise<void>,
+    supportsFinalisation?: boolean,
+  ): Promise<number | undefined> {
+    this.supportsFinalization = supportsFinalisation;
+    return super.init(reindex);
+  }
+
+  // Detect a fork by walking back through unfinalized blocks
+  @profiler()
```

Comment:
better to have some explanation in comment what the returned header is.
without looking into the code, I will think the return is the forked (falsy) header.